### PR TITLE
Use DialTLSContext instead of DialTLS

### PR DESCRIPTION
**Pull request by jiuker**
**Please provide a description of this PR:**

Use DialTLSContext instead of DialTLS

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Can you help explain the benefits of this? AFAIK there is no difference.

**Comment by jiuker** [comment]
> Can you help explain the benefits of this? AFAIK there is no difference.

This property is deprecated

**Comment by zhlsunshine** [comment]
> > Can you help explain the benefits of this? AFAIK there is no difference.
> 
> This property is deprecated

I suggest you should change them all together. I notice that it's mixed in using both `DialTLS` and `DialTLSContext` in some places.

**Comment by jiuker** [comment]
> > > Can you help explain the benefits of this? AFAIK there is no difference.
> > 
> > 
> > This property is deprecated
> 
> I suggest you should change them all together. I notice that it's mixed in using both `DialTLS` and `DialTLSContext` in some places.

Yes, there is also this attribute inside another file. I have modified it. review plz. @zhlsunshine

**Comment by zhlsunshine** [comment]
> > > > Can you help explain the benefits of this? AFAIK there is no difference.
> > > 
> > > 
> > > This property is deprecated
> > 
> > 
> > I suggest you should change them all together. I notice that it's mixed in using both `DialTLS` and `DialTLSContext` in some places.
> 
> Yes, there is also this attribute inside another file. I have modified it. review plz. @zhlsunshine

LGTM

File: releasenotes/notes/42580.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: promotion
+area: istioctl
+releaseNotes:
+  - |
+    **Fixed** Use Context instead of noctx.
```
Comment (ymesika) [inline releasenotes/notes/42580.yaml:6]: Not sure a release note is indeed necessary as it's internal but maybe mention the replacement of a deprecated `DialTLS` with `DialTLSContext`?

Comment (jiuker) [reply releasenotes/notes/42580.yaml:6]: @ymesika It doesn't seem necessary. This is a test file.

Comment (ymesika) [reply releasenotes/notes/42580.yaml:6]: You can remove this file now with the `release-notes-none` label.

**Review by howardjohn — CHANGES_REQUESTED** [review]
There is no benefits of this change. Its deprecated because you are supposed to actually *use* the context; Since we don't care its more appropriate to use the old DialTLS function. See https://github.com/istio/istio/wiki/Development-Conventions#superfluous-changes

**Pull request closed without merging**